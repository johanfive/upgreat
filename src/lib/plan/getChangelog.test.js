const axios = require('axios')
jest.mock('axios')

const getChangelog = require('./getChangelog')

it('should return null if repo url cant be resolved', async () => {
  await expect(getChangelog('')).resolves.toMatchSnapshot()
})

it('should return CHANGELOG.md url if it exists', async () => {
  axios.get.mockImplementationOnce(() => Promise.resolve())

  await expect(
    getChangelog('https://github.com/facebook/jest'),
  ).resolves.toMatchSnapshot()
})

it('should return changelog.md url if it exists', async () => {
  axios.get.mockImplementationOnce(() => Promise.reject())
  axios.get.mockImplementationOnce(() => Promise.resolve())

  await expect(
    getChangelog('https://github.com/facebook/jest'),
  ).resolves.toMatchSnapshot()
})

it('should return null if changelog not found', async () => {
  axios.get.mockImplementation(() => Promise.reject())

  await expect(
    getChangelog('https://github.com/facebook/jest'),
  ).resolves.toMatchSnapshot()
})
