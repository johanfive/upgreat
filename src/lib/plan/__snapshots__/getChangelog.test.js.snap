// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should return CHANGELOG.md url if it exists 1`] = `
Array [
  "https://github.com/facebook/jest.git",
  "https://raw.githubusercontent.com/facebook/jest/master/CHANGELOG.md",
]
`;

exports[`should return changelog.md url if it exists 1`] = `
Array [
  "https://github.com/facebook/jest.git",
  "https://raw.githubusercontent.com/facebook/jest/master/changelog.md",
]
`;

exports[`should return null if changelog not found 1`] = `
Array [
  "https://github.com/facebook/jest.git",
  null,
]
`;

exports[`should return null if repo url cant be resolved 1`] = `
Array [
  null,
  null,
]
`;
